ovs/extensions/healthcheck/utils/extension.py

```diff
@@ -184,8 +184,19 @@ def check_etcd(self):
 
         @rtype: bool
         """
-
-        result = self.execute_bash_command("dpkg -l | grep etcd")
+        
+        #add check os-release version
+        try:
+            release_version_return = self.execute_bash_command('cat /etc/os-release')
+            # All OS distribution classes used in below code should share the exact same interface!
+            if 'CentOS Linux' in release_version_return[0]:
+                result = self.execute_bash_command("rpm -qa | grep etcd")
+            elif 'Ubuntu' in release_version_return[0]:
+                result = self.execute_bash_command("dpkg -l | grep etcd")
+            else:
+                raise RuntimeError('The OS is not supported')
+        except Exception as ex:
+            raise RuntimeError('Error checking etcd: {0}'.format(ex))
```

Comment:
Better not catch the exception, but let it fall through so we don't lose stacktrace information.